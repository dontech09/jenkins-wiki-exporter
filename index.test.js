/* eslint-env node, jest */
const sut = require('./index.js');
const MockExpressRequest = require('mock-express-request');
const MockExpressResponse = require('mock-express-response');
const mockWikiPrefix = 'https://wiki.jenkins.io/display/JENKINS/';
jest.mock('./confluence.js', () => ({
  ...require.requireActual('./confluence.js'),
  getRawConfluenceContent: (url) => {
    const pluginName = url.replace(mockWikiPrefix, '');
    return require('fs').promises.readFile(`__testData/${pluginName}.json`)
        .then((buf) => JSON.parse(buf.toString()).wiki.content);
  },
}));

jest.mock('./reports.js', () => ({
  ...require.requireActual('./reports.js'),
  getPluginWikiUrl: (pluginName) => {
    return `${mockWikiPrefix}${pluginName}`;
  },
}));

describe('/plugin/:pluginName', function() {
  it('handle the basics', async () => {
    const req = {
      log: {
        debug: jest.fn(),
      },
      params: {
        plugin: 'html5-notifier-plugin',
      },
    };
    const res = {
      send: jest.fn(),
      type: jest.fn(),
    };

    await sut.requestPluginHandler(req, res);
    expect(res.send.mock.calls).toMatchSnapshot();
  });

  it('markdown should return markdown', async () => {
    const req = {
      log: {
        debug: jest.fn(),
      },
      params: {
        plugin: 'html5-notifier-plugin',
        format: 'md',
      },
    };
    const res = {
      send: jest.fn(),
      type: jest.fn(),
    };

    await sut.requestPluginHandler(req, res);
    expect(res.send.mock.calls).toMatchSnapshot();
  });
  it('markdown.zip should return markdown and images in zip', async () => {
    const req = new MockExpressRequest({
      log: {
        debug: jest.fn(),
      },
      params: {
        plugin: 'html5-notifier-plugin',
        format: 'md.zip',
      },
    });
    const res = new MockExpressResponse({ });

    await sut.requestPluginHandler(req, res);
    expect(res._getString()).toMatchSnapshot();
  });
  it('adoc should return asciidoc', async () => {
    const req = {
      log: {
        debug: jest.fn(),
      },
      params: {
        plugin: 'html5-notifier-plugin',
        format: 'adoc',
      },
    };
    const res = {
      send: jest.fn(),
      type: jest.fn(),
    };

    await sut.requestPluginHandler(req, res);
    expect(res.send.mock.calls).toMatchSnapshot();
  });
});

