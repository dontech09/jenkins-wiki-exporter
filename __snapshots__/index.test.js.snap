// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/plugin/:pluginName handle the basics 1`] = `
Array [
  Array [
    "# HTML5 Notifier Plugin

  

Provides [W3C Web
Notifications](http://dev.w3.org/2006/webapi/WebNotifications/publish/Notifications.html)
support for builds.

### Browser compatibility

This plugin is compatible with Google Chrome, Safari, and Firefox

Other browsers do not support HTML5 notifications right now. You can
check this website for details
: <http://caniuse.com/#feat=notifications>

### Screenshots

|                      |                                                                                                                                             |
|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
| Safari Notification  | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-notifications-1.3.png?version=1&modificationDate=1407001198000&api=v2) |
| System Configuration | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-configure-1.1.png?version=1&modificationDate=1328215062000&api=v2)     |
| Job Configuration    | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-job-configure-1.1.png?version=1&modificationDate=1328215063000&api=v2) |

  

------------------------------------------------------------------------

  

### Changelog

#### 1.5 (released 2015-03-06)

-   [JENKINS-27222](https://issues.jenkins-ci.org/browse/JENKINS-27222) -
    Fixes global configuration not saving/loading.

#### 1.3 (released 2014-08-01)

-   [JENKINS-16236](https://issues.jenkins-ci.org/browse/JENKINS-16236) -
    upgrade to use current html5 notification spec. Removes html in the
    notifications but now works again.

#### 1.2 (released 2012-04-06)

-   updated minimum required core to v1.455 to pick up prototype.js:1.7
-   [JENKINS-13138](https://issues.jenkins-ci.org/browse/JENKINS-13138) -
    Verify action doesn't work with CSRF option
-   [JENKINS-13322](https://issues.jenkins-ci.org/browse/JENKINS-13322) -
    Configure tab freezes when using both the HTML5 notifier plugin and
    the Android Emulator Plugin
-   [JENKINS-12538](https://issues.jenkins-ci.org/browse/JENKINS-12538) -
    java.lang.String cannot be cast to net.sf.json.JSONArray
-   [JENKINS-13038](https://issues.jenkins-ci.org/browse/JENKINS-13038) -
    HTML5 notifier plugin breaks Jenkins with CSRF protection

#### 1.1 (released 2012-02-02)

-   JENKINS-11618 - prototype.js
-   added per-job job property to skip notifications
-   added ability to only notify on different build result than previous
-   cleaned up the notification area html

#### 1.0 (released 2011-10-19)

-   initial implementation
-   fully I18N compliant
",
  ],
]
`;

exports[`/plugin/:pluginName markdown should return markdown 1`] = `
Array [
  Array [
    "# HTML5 Notifier Plugin

  

Provides [W3C Web
Notifications](http://dev.w3.org/2006/webapi/WebNotifications/publish/Notifications.html)
support for builds.

### Browser compatibility

This plugin is compatible with Google Chrome, Safari, and Firefox

Other browsers do not support HTML5 notifications right now. You can
check this website for details
: <http://caniuse.com/#feat=notifications>

### Screenshots

|                      |                                                                                                                                             |
|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
| Safari Notification  | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-notifications-1.3.png?version=1&modificationDate=1407001198000&api=v2) |
| System Configuration | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-configure-1.1.png?version=1&modificationDate=1328215062000&api=v2)     |
| Job Configuration    | ![](https://wiki.jenkins.io/download/attachments/58917110/screenshot-job-configure-1.1.png?version=1&modificationDate=1328215063000&api=v2) |

  

------------------------------------------------------------------------

  

### Changelog

#### 1.5 (released 2015-03-06)

-   [JENKINS-27222](https://issues.jenkins-ci.org/browse/JENKINS-27222) -
    Fixes global configuration not saving/loading.

#### 1.3 (released 2014-08-01)

-   [JENKINS-16236](https://issues.jenkins-ci.org/browse/JENKINS-16236) -
    upgrade to use current html5 notification spec. Removes html in the
    notifications but now works again.

#### 1.2 (released 2012-04-06)

-   updated minimum required core to v1.455 to pick up prototype.js:1.7
-   [JENKINS-13138](https://issues.jenkins-ci.org/browse/JENKINS-13138) -
    Verify action doesn't work with CSRF option
-   [JENKINS-13322](https://issues.jenkins-ci.org/browse/JENKINS-13322) -
    Configure tab freezes when using both the HTML5 notifier plugin and
    the Android Emulator Plugin
-   [JENKINS-12538](https://issues.jenkins-ci.org/browse/JENKINS-12538) -
    java.lang.String cannot be cast to net.sf.json.JSONArray
-   [JENKINS-13038](https://issues.jenkins-ci.org/browse/JENKINS-13038) -
    HTML5 notifier plugin breaks Jenkins with CSRF protection

#### 1.1 (released 2012-02-02)

-   JENKINS-11618 - prototype.js
-   added per-job job property to skip notifications
-   added ability to only notify on different build result than previous
-   cleaned up the notification area html

#### 1.0 (released 2011-10-19)

-   initial implementation
-   fully I18N compliant
",
  ],
]
`;

exports[`/plugin/:pluginName adoc should return asciidoc 1`] = `
Array [
  Array [
    "[[HTML5NotifierPlugin-HTML5NotifierPlugin]]
== HTML5 Notifier Plugin

 +

[.conf-macro .output-inline]#Provides
http://dev.w3.org/2006/webapi/WebNotifications/publish/Notifications.html[W3C
Web Notifications] support for builds.#

[[HTML5NotifierPlugin-Browsercompatibility]]
=== Browser compatibility

This plugin is compatible with Google Chrome, Safari, and Firefox

Other browsers do not support HTML5 notifications right now. You can
check this website for details : http://caniuse.com/#feat=notifications

[[HTML5NotifierPlugin-Screenshots]]
=== Screenshots

[cols=\\",\\",]
|===
|Safari Notification
|[.confluence-embedded-file-wrapper]#image:https://wiki.jenkins.io/download/attachments/58917110/screenshot-notifications-1.3.png?version=1&modificationDate=1407001198000&api=v2[image]#

|System Configuration
|[.confluence-embedded-file-wrapper]#image:https://wiki.jenkins.io/download/attachments/58917110/screenshot-configure-1.1.png?version=1&modificationDate=1328215062000&api=v2[image]#

|Job Configuration
|[.confluence-embedded-file-wrapper]#image:https://wiki.jenkins.io/download/attachments/58917110/screenshot-job-configure-1.1.png?version=1&modificationDate=1328215063000&api=v2[image]#
|===

 +

'''''

 +

[[HTML5NotifierPlugin-Changelog]]
=== Changelog

[[HTML5NotifierPlugin-1.5(released2015-03-06)]]
==== 1.5 (released 2015-03-06)

* https://issues.jenkins-ci.org/browse/JENKINS-27222[JENKINS-27222] -
Fixes global configuration not saving/loading.

[[HTML5NotifierPlugin-1.3(released2014-08-01)]]
==== 1.3 (released 2014-08-01)

* https://issues.jenkins-ci.org/browse/JENKINS-16236[JENKINS-16236] -
upgrade to use current html5 notification spec. Removes html in the
notifications but now works again.

[[HTML5NotifierPlugin-1.2(released2012-04-06)]]
==== 1.2 (released 2012-04-06)

* updated minimum required core to v1.455 to pick up prototype.js:1.7
* https://issues.jenkins-ci.org/browse/JENKINS-13138[JENKINS-13138] -
Verify action doesn't work with CSRF option
* https://issues.jenkins-ci.org/browse/JENKINS-13322[JENKINS-13322] -
Configure tab freezes when using both the HTML5 notifier plugin and the
Android Emulator Plugin
* https://issues.jenkins-ci.org/browse/JENKINS-12538[JENKINS-12538] -
java.lang.String cannot be cast to net.sf.json.JSONArray
* https://issues.jenkins-ci.org/browse/JENKINS-13038[JENKINS-13038] -
HTML5 notifier plugin breaks Jenkins with CSRF protection

[[HTML5NotifierPlugin-1.1(released2012-02-02)]]
==== 1.1 (released 2012-02-02)

* JENKINS-11618 - prototype.js
* added per-job job property to skip notifications
* added ability to only notify on different build result than previous
* cleaned up the notification area html

[[HTML5NotifierPlugin-1.0(released2011-10-19)]]
==== 1.0 (released 2011-10-19)

* initial implementation
* fully I18N compliant
",
  ],
]
`;

exports[`/plugin/:pluginName markdown.zip should return markdown and images in zip 1`] = `""`;
